apps/src/lib/node/ledger/ethereum_node/oracle.rs

```diff
@@ -57,11 +77,14 @@ impl Oracle {
         url: &str,
         sender: BoundedSender<EthereumEvent>,
         abort: Sender<()>,
+        backoff: Duration,
     ) -> Self {
         Self {
             client: Web3::new(url, std::time::Duration::from_secs(30)),
             sender,
             abort: Some(abort),
+            backoff,
+            blocks_processed: None,
```

Comment:
Essentially, this is what bothers me. We have a test artifact in our production code. If you don't like the idea of PR #666, then we better make an Oracle trait so that we can make a mock oracle for tests. The only method that would differ is `run_oracle_aux`.

Reply:
I'd like to do this as well - have opened semi-related https://github.com/anoma/namada/issues/672 to fix up our `web30` dependency so that it should be easier to switch the `Oracle` to using a trait in the future